### PR TITLE
Redux setup and data persist

### PR DESCRIPTION
**Pull request by NancyRike**

### CODE REVIEW

File: src/reduxStore/Reducers/authReducer.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+interface IAction {
+    type:string,
+    payload:any
+}
+
+ const AuthReducer = (state= initialState, action:IAction) => {
+        switch (action.type) {
+            case IS_AUTHENTICATED:
+                return{...state, isAuthenticated: action.type }
```
Comment (Dverybest) [inline src/reduxStore/Reducers/authReducer.ts:12]: Can we combine this two
to
            case USER_DETAILS:
                return{...state,isAuthenticated:true, userDetails: action.type}

File: src/reduxStore/Reducers/authReducer.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            case USER_DETAILS:
+                return{...state, userDetails: action.type}
+            case TOKEN:
+                return {...state, token: action.type}
```
Comment (Dverybest) [inline src/reduxStore/Reducers/authReducer.ts:16]: you should not  action.type rather action.payload
this should be
 case USER_DETAILS:
                return{...state, userDetails: action.payload}
            case TOKEN:
                return {...state, token: action.payload}

File: src/reduxStore/actions/authAction.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import {LOGIN_AUTH} from "../constant";
+
+interface payloadType {
+    type:any,
+}
+
+const AuthAction = (payload:payloadType)=>{
```
Comment (Dverybest) [inline src/reduxStore/actions/authAction.ts:7]: can you change this to
const AuthAction = (payload:any)=>{

File: src/reduxStore/actions/authAction.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import {LOGIN_AUTH} from "../constant";
+
+interface payloadType {
```
Comment (Dverybest) [inline src/reduxStore/actions/authAction.ts:3]: No need for this

File: src/reduxStore/store.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React, { ReactElement } from 'react';
+import { createStore, applyMiddleware } from "redux";
+import RootReducer from "./Reducers/RootReducer";
+import thunk from "redux-thunk";
+interface Props {
+
+}
+
+ const Store = ()=> {
+    return createStore(RootReducer, applyMiddleware (thunk));
```
Comment (Dverybest) [inline src/reduxStore/store.js:10]: Add middleware to persist data to sessionStorage

**Review by Dverybest — CHANGES_REQUESTED** [review]

File: src/reduxStore/Reducers/authReducer.ts (hunk 1 above)
Comment (NancyRike) [reply src/reduxStore/Reducers/authReducer.ts:12]: okay will do

**Review by Dverybest — APPROVED** [review]

**Pull request merged**